client/client.go

```diff
@@ -206,6 +207,16 @@ func (c *LogClient) queueLeaf(ctx context.Context, leaf *trillian.LogLeaf) error
 		LogId: c.LogID,
 		Leaf:  leaf,
 	}
-	_, err := c.client.QueueLeaf(ctx, &req)
-	return err
+	rsp, err := c.client.QueueLeaf(ctx, &req)
+	if err != nil {
+		return err
+	}
+	if rsp == nil {
+		return errors.New("no QueueLeafResponse generated")
+	}
+	if rsp.QueuedLeaf.Status.Code == int32(code.Code_ALREADY_EXISTS) {
```

Comment:
The codes in `rpc/code` should be exactly the same a the ones in `grpc/codes` Might save a type conversion.